File: src/test/java/com/soapdogg/lispInterpreter/parser/ParserResultBuilderTest.java
package com.soapdogg.lispInterpreter.parser;

import com.soapdogg.lispInterpreter.datamodels.Node;
import com.soapdogg.lispInterpreter.datamodels.Token;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import java.util.ArrayList;
import java.util.List;

public class ParserResultBuilderTest {

    private Node resultingNode;
    private List<Token> remainingTokens;

    private ParserResultBuilder parserResultBuilder;

    @BeforeEach
    void setup() {
        resultingNode = Mockito.mock(Node.class);
        remainingTokens = new ArrayList<>();

        parserResultBuilder = new ParserResultBuilder();
    }

    @Test
    void buildParserResultTest() {
        final var actual = parserResultBuilder.buildParserResult(
            resultingNode,
            remainingTokens
        );

        Assertions.assertEquals(resultingNode, actual.getResultingNode());
        Assertions.assertEquals(remainingTokens, actual.getRemainingTokens());
    }
}

File: src/test/java/com/soapdogg/lispInterpreter/parser/ParserResultBuilderTest.kt
package com.soapdogg.lispInterpreter.parser

import com.soapdogg.lispInterpreter.datamodels.Node
import com.soapdogg.lispInterpreter.datamodels.Token
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

class ParserResultBuilderTest {
    private val parserResultBuilder = ParserResultBuilder()

    @Test
    fun buildParserResultTest() {
        val resultingNode = Mockito.mock(Node::class.java)
        val remainingTokens: List<Token> = emptyList()
        val (resultingNode1, remainingTokens1) = parserResultBuilder.buildParserResult(
            resultingNode,
            remainingTokens
        )
        Assertions.assertEquals(resultingNode, resultingNode1)
        Assertions.assertEquals(remainingTokens, remainingTokens1)
    }
}